core/feed_protocol/opds.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Any, Optional
+
+from core.feed_protocol.base import FeedProtocol
+from core.feed_protocol.serializer.opds import OPDS1Serializer
+from core.feed_protocol.types import FeedData, WorkEntry
+from core.util.flask_util import OPDSEntryResponse, OPDSFeedResponse
+
+if TYPE_CHECKING:
+    pass
+
+
+class OPDSFeedProtocol(FeedProtocol):
+    def __init__(
+        self, title: str, url: str, precomposed_entries: Optional[Any] = None
```

Comment:
How does `precomposed_entries` entries work, when this class is version agnostic. This parameter is what prompted me to ask about type hinting the class, because following through the logic I had a hard time understanding how this would work, it seem like these entries are strongly tied to the atom feed class. The type hint here of `Optional[Any]`, didn't give me the clarity I was hoping for though 😄

Reply:
The type hint in the inherited class is `Optional[List[OPDSMessage]]`, I looked into the previous usage and it seemed like the only type being used.
The [OPDS1.x serializer](https://github.com/ThePalaceProject/circulation/pull/1308/files/ceabc274dfe01c8472ff2162ee266db0507ebb5b#diff-f0118b812f75a0e1c1a2cfa89565a009f1f2d554e5bb14d52b265bec044281dcR70) just uses it's `.tag` method, where as the OPDS2 serializer does not implement these in #1340 , since I wasn't sure what format it should translate into.

Reply:
Okay. Sounds good. We can iterate on this more in #1340. I'll close this one.